### PR TITLE
pom.xml updates

### PR DESCRIPTION
**Pull request by ashitsalesforce**
- speed up jar building by skipping dmg creation.

- sign jar using mvn target and jarsigner plug-in.

- fix mvn warning about not specifying the version for maven-compiler-plugin.

- include Windows-specific SWT DLLs in the jar if Windows build target is specified.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -221,6 +223,13 @@
         <directory>src/main/nsis/bin</directory>
         <filtering>true</filtering>
       </resource>
+      <resource>
+        <directory>${DLLFolder}</directory>
+        <filtering>false</filtering>
```
Comment (jjangsfdc) [inline pom.xml:228]: What does this filtering=false do?

File: pom.xml [hunk 2]
```diff
@@ -401,11 +411,29 @@
...
+          <execution>
+            <id>sign</id>
+            <goals>
+              <goal>sign</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <archive>target/dataloader-${project.version}-uber.jar</archive>
+          <skip>${jarsigner.skip}</skip>
```
Comment (jjangsfdc) [inline pom.xml:434]: Looking at line 29, the variable is set to 'false' then will it not sign the Jar? Also, how will this work for users who would want to build it on their machine without actually signing?

**Review by jjangsfdc — COMMENTED** [review]
As per yesterday's discussion, did you also incorporated the fix of changing one place to update the whole version?

File: pom.xml (hunk 1 above)
Comment (ashitsalesforce) [reply pom.xml:228]: From https://books.sonatype.com/mvnref-book/reference/resource-filtering-sect-description.html:

Using Maven resource filtering you can reference Maven properties and then use Maven profiles to define different configuration values for different target deployment environments. 

Given that we are not referencing any Maven properties for including DLLs for Windows environment, I set <filtering> to false.

File: pom.xml (hunk 2 above)
Comment (ashitsalesforce) [reply pom.xml:434]: The variable is set to true in pom.xml, thereby skipping jar signing by default. So, anyone who wants to build without signing should run "mvn clean package -DskipTests".

**Pull request merged**